ego/cli/bundle.go

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package cli
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"debug/elf"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// ErrFileIsNotAnEGoBinary is returned when the file is not an enclave, therefore cannot possibly be bundled.
+var ErrFileIsNotAnEGoBinary = errors.New("file is not an EGo Binary")
+
+// ErrFileIsAlreadyBundled is returned when the file is already bundled. Technically, the outer binary is not an enclave either, but this is a more specific error.
+var ErrFileIsAlreadyBundled = errors.New("file is already bundled")
+
+// ErrFileIsNotAnEGoBinary is returned when the enclave has not been signed yet with 'ego sign', which would create a non-functional bundle.
+var ErrFileHasNotBeenSignedYet = errors.New("enclave has not been signed yet with 'ego sign'")
+
+// Bundle bundles an enclave with the currently installed EGo runtime into a single executable.
+func (c *Cli) Bundle(filename string) error {
+	// Check if the file is an enclave or already bundled
+	if err := c.checkIfBundable(filename); err != nil {
+		return err
+	}
+
+	// Build runtime image
+	runtimePath, err := c.buildImage(filename)
```

Comment:
the names doesn't seem to fit well now that it also bundles the enclave itself in the tar